Source/WebGPU/WGSL/TypeStore.h

```diff
@@ -60,6 +61,18 @@ class TypeStore {
     Type* constructType(AST::ParameterizedTypeName::Base, Type*);
 
 private:
+    class TypeCache {
+    public:
+        template<typename Key>
+        Type* find(const Key&) const;
+
+        template<typename Key>
+        void insert(const Key&, Type*);
+
+    private:
+        HashMap<std::tuple<Type*, uint64_t>, Type*> m_storage;
```

Comment:
Mapping a `Type*` + extra magic number to `Type*` seems a bit strange to me. Is this how it's done in JSC?

Reply:
I don't know if we do that specifically in JSC, but it's pretty common to serialize data into unsigned types. I think this gives us enough information to put all the types in the same HashMap, avoid the overhead of having one HashMap for each kind of type.

Reply:
I'll add a comment here to explain my thoughts and the layout of the second word